### DISCUSSION TITLE
Read Lakehouse data with SQL Endpoint using Entra ID without adding in each workspace as Member/Contributor

### DISCUSSION THREAD

**Discussion by SanchanaMohankumar** [microsoft/Microsoft-Fabric-workload-development-sample#29] [Q&A] [answered]
Hi Team, 
In our scenario, we are attempting to read the table records using an SQL connection string instead of the Lakehouse file path. Currently, we use a Service Principal to connect to the backend, and to connect to Microsoft Fabric, we add the Entra ID to each workspace to access the SQL Endpoint connection string. Why is this necessary, and is this a feasible long-term solution? Your input on this matter would be greatly appreciated.
Thank you

**Comment by ramizsha** [comment, marked as answer]
This is the Fabric security model. Workspace is the top-level object and user(s) must be granted access to it to access the data it contains.
